src/main/java/com/opentable/kafka/builders/KafkaBuilderFactoryBean.java

```diff
@@ -83,7 +84,7 @@
         );
         final KafkaConsumerBuilder<?, ?> res = new KafkaConsumerBuilder<>(mergedSeedProperties, environmentProvider);
         metricRegistry.ifPresent(res::withMetricRegistry);
-        serviceInfo.ifPresent(si -> res.withClientId(name + "-" + si.getName()));
+        serviceInfo.ifPresent(si -> res.withClientId(name + "-" + si.getName()  + "-" + ClientIdGenerator.getInstance().nextClientId()));
```

Comment:
The problem with this is the order of initialization isn't guaranteed in beans. So things will flop around and alarms will get bad. I really think people have to take responsibiliuty for this themselves.